### PR TITLE
Enable native GQA and hoist mask computation for Metal SDPA

### PR DESCRIPTION
**Pull request by mergennachin**
Replace F.scaled_dot_product_attention with a direct call to
torch.ops.aten._scaled_dot_product_attention_math_for_mps in
StandardSDPA. This is necessary because F.scaled_dot_product_attention
is CompositeImplicitAutograd — torch.export() decomposes it back into
repeat_interleave + matmul for GQA, defeating native kernel support.
The _for_mps op stays as a single node in the exported graph and
resolves at runtime to the custom Metal SDPA shader in op_sdpa.mm,
which handles GQA natively via gqa_factor = n_heads / n_kv_heads.

For voxtral (32 Q heads, 8 KV heads), this eliminates 4x redundant
K/V memory traffic per layer (repeat_interleave materialized 128MB
of expanded KV per layer vs 32MB with native GQA).

Also hoist the attention mask computation from StandardSDPA (called
26x per token, once per layer) to MistralDecoder.forward (called 1x).
Build the mask using integer arithmetic (clamp) instead of bool
comparisons since Metal AOTI doesn't support bool tensor allocation.

The XNNPACK path is unchanged.

Improved from 19 tokens/s to 40 tokens/s

Test Plan: 

```
I 00:00:26.888453 executorch:voxtral_realtime_runner.cpp:247] Audio: 314240 samples -> 375 frames
 One, two, three. This is February 18th, Wednesday, and I'm testing an AI model. Tell me an interesting story. One, two, three. Thank you..</s>
PyTorchObserver {"prompt_tokens":0,"generated_tokens":378,"model_load_start_ms":1772062281912,"model_load_end_ms":1772062303509,"inference_start_ms":1772062303509,"inference_end_ms":1772062319075,"prompt_eval_end_ms":1772062309797,"first_token_ms":1772062309797,"aggregate_sampling_time_ms":0,"SCALING_FACTOR_UNITS_PER_SECOND":1000}
I 00:00:37.164707 executorch:stats.h:143] 	Prompt Tokens: 0    Generated Tokens: 378
I 00:00:37.164710 executorch:stats.h:149] 	Model Load Time:		21.597000 (seconds)
I 00:00:37.164713 executorch:stats.h:159] 	Total inference time:		15.566000 (seconds)		 Rate: 	24.283695 (tokens/second)
I 00:00:37.164715 executorch:stats.h:167] 		Prompt evaluation:	6.288000 (seconds)		 Rate: 	0.000000 (tokens/second)
I 00:00:37.164736 executorch:stats.h:178] 		Generated 378 tokens:	9.278000 (seconds)		 Rate: 	40.741539 (tokens/second)
I 00:00:37.164747 executorch:stats.h:186] 	Time to first generated token:	6.288000 (seconds)
I 00:00:37.164749 executorch:stats.h:193] 	Sampling time over 378 tokens:	0.000000 (seconds)
I 00:00:37.166004 executorch:metal_backend.cpp:716] Removed temporary shared library file: /var/folders/_1/z_wzgpv50gn73c02j5xmcnf40000gn/T/text_decoder_so_blob46972.so
I 00:00:37.291288 executorch:memory.cpp:642] Cleared all tensors and Metal resources
I 00:00:37.291937 executorch:metal_backend.cpp:716] Removed temporary shared library file: /var/folders/_1/z_wzgpv50gn73c02j5xmcnf40000gn/T/token_embedding_so_blob46972.so
I 00:00:37.291944 executorch:memory.cpp:642] Cleared all tensors and Metal resources
I 00:00:37.292499 executorch:metal_backend.cpp:716] Removed temporary shared library file: /var/folders/_1/z_wzgpv50gn73c02j5xmcnf40000gn/T/audio_encoder_so_blob46972.so
I 00:00:37.292507 executorch:memory.cpp:642] Cleared all tensors and Metal resources
```

### CODE REVIEW

File: examples/models/voxtral_realtime/model.py [hunk 1]
```diff
@@ -427,11 +427,33 @@ def forward(
         return y.view(bsz, seqlen, self.dim).to(dtype=input_dtype)
 
 
+def _build_attn_mask(
+    input_pos: torch.Tensor, max_seq_len: int, device: torch.device
+) -> torch.Tensor:
+    """Build float additive attention mask without bool intermediates.
+
+    Metal AOTI doesn't support bool tensor allocation on MPS, so we use
```
Comment (manuelcandales) [inline examples/models/voxtral_realtime/model.py:435]: Would it run faster if I enable this?

File: examples/models/voxtral_realtime/model.py [hunk 2]
```diff
@@ -549,7 +543,10 @@ def forward(
 
         k, v = self.kv_cache.update(input_pos, k, v)
 
-        y = self.sdpa(input_pos, q, k, v, B, T)
+        if self.use_standard_attention:
```
Comment (manuelcandales) [inline examples/models/voxtral_realtime/model.py:546]: change `use_standard_attention` name to something more specific to Metal, since this is now using a specific MetalSDPA

**Review by manuelcandales — APPROVED** [review]

File: examples/models/voxtral_realtime/model.py (hunk 2 above)
Comment (mergennachin) [reply examples/models/voxtral_realtime/model.py:546]: Yeah, I will change in a follow-up PR

File: examples/models/voxtral_realtime/model.py (hunk 1 above)
Comment (mergennachin) [reply examples/models/voxtral_realtime/model.py:435]: This didn't make a big difference based on my local profiling
